api/tests/routes/backoffice_v3/collective_offer_templates_test.py

```diff
@@ -389,3 +393,114 @@ def test_get_reject_form_test(self, legit_user, authenticated_client):
         with assert_num_queries(2):  # session + user
             response = authenticated_client.get(form_url)
             assert response.status_code == 200
+
+
+class BatchCollectiveOfferTemplatesValidateTest(PostEndpointHelper):
+    endpoint = "backoffice_v3_web.collective_offer_template.batch_validate_collective_offer_templates"
+    needed_permission = perm_models.Permissions.FRAUD_ACTIONS
+
+    def test_batch_validate_collective_offer_templates(self, legit_user, authenticated_client):
+        collective_offer_templates = educational_factories.CollectiveOfferTemplateFactory.create_batch(
+            3, validation=OfferValidationStatus.PENDING
+        )
+        parameter_ids = ", ".join(
+            str(collective_offer_template.id) for collective_offer_template in collective_offer_templates
+        )
+        response = self.post_to_endpoint(authenticated_client, form={"object_ids": parameter_ids})
+
+        assert response.status_code == 303
+
+        expected_url = url_for(
+            "backoffice_v3_web.collective_offer_template.list_collective_offer_templates", _external=True
+        )
+        assert response.location == expected_url
+
+        for collective_offer_template in collective_offer_templates:
+            db.session.refresh(collective_offer_template)
+            assert collective_offer_template.lastValidationDate.strftime("%d/%m/%Y") == datetime.date.today().strftime(
+                "%d/%m/%Y"
+            )
+            assert collective_offer_template.isActive is True
+            assert collective_offer_template.lastValidationType is OfferValidationType.MANUAL
+            assert collective_offer_template.validation is OfferValidationStatus.APPROVED
+
+        assert len(mails_testing.outbox) == 3
+        for i, email in enumerate(mails_testing.outbox):
+            assert email.sent_data["To"] == collective_offer_templates[i].venue.bookingEmail
```

Comment:
Comme la requête avec le critère `.CollectiveOfferTemplate.id.in_(collective_offer_template_ids)` ne garantit pas les lignes dans l'ordre des ids, ce test sera probablement instable, en traitant les offres dans un ordre non déterministe, et donc envoyant les emails dans ce même ordre qu'on ne peut prédire.

Dans ce cas, préférer plutôt : 
`assert {email.sent_data["To"] for email in mails_testing.outbox} == {template[i].venue.bookingEmail for template in collective_offer_templates}`